students/erik_oien/lesson05/mailroom3.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+import sys 
+import os
+
+donor_dict = {
+    "donor1": {"first_name": "Nathan", "last_name": "Explosion", "donations": [39595, 35081, 93295]}, 
+    "donor2": {"first_name": "Skwisgaar", "last_name": "Skwigelf", "donations": [37198]}, 
+    "donor3": {"first_name": "Toki", "last_name": "Wartooth", "donations": [20037, 32892, 99788]},
+    "donor4": {"first_name": "William", "last_name": "Murderface", "donations": [87470, 86870, 4397]},
+    "donor5": {"first_name": "Pickles", "last_name": "", "donations": [87838, 60282, 26653]},
+}
+
+# main menu for program 
+
+def menu():
+    while True:
+        welcome_input = input(opening_promt)
+        try:
+            main_menu_switch[welcome_input]()
+        except(KeyError):
+            print("Invalid option, please choose a number from the list below:")
+
+# thank you menu
+def thank_you():
+    while True:
+        donor_name = input(f"Please choose a donor or type list to see previous donors.\n{sub_menu_prompt}")
+        past_donor = False
+        if donor_name.lower() in ["menu", "quit"]:
+            sub_menu_switch[donor_name.lower()]()
+        elif donor_name.lower() == 'list':
+            past_donor = False
+            for donor in donor_dict.items():
+                print(donor[1]["first_name"] + " " + donor[1]["last_name"])
+        else:
+            try:
+                for donor in donor_dict.items():
+                    if donor_name.lower() == (donor[1]["first_name"].lower() + " " + donor[1]["last_name"].lower()).strip():
+                        past_donor = True
+                        donation_amount = input("How much was the most recent donation from {}? >>> ".format(donor_name))
+                        donor[1]["donations"].append(int(donation_amount))
+                if past_donor == False:
+                    print("Looks like this is a new donor, lets add them!")
+                    first_name, last_name = donor_name.split(" ", 1)
+                    donation_amount = input("How much did {} donate? >>> ".format(donor_name))
+                    new_donor = {"first_name": first_name, "last_name": last_name, "donations": [int(donation_amount)]}
+                    donor_dict[f"donor{len(donor_dict)+1}"] = new_donor
+                email = thank_you_email(donor_name, donation_amount)
+                print(email)
+                return False
+            except(ValueError):
+                print("Expecting first and last name, only one name entered")
+                
+# create report menu
+
+def create_report():
+    while True:
+        donor_report = []
+        [donor_report.append(donor_info(donor)) for donor in donor_dict.items()]
+        donor_report_info = sorted(donor_report, key=sort_key, reverse=True)
+        table_header()
+        [row_formatter(donor) for donor in donor_report_info]
```

Comment:
even though this lesson is about trying to use comprehensions - you actually want to avoid this pattern.

the issue here is that this return a list of None values, since `row_formatter` has no return value and you're not storing the final result anyway.

using a regular for loop here is a better practice.